corehq/apps/case_search/tests/test_case_search_filters.py

```diff
@@ -221,3 +221,46 @@ def test_match_all(self):
             None,
             ['c1', 'c2', 'c3']
         )
+
+    def test_OR_two_postiive_conditions(self):
+        self._create_case_search_config()
+        cases = [
+            {'_id': 'c1', 'case_type': 'wizard', 'first_name': 'Harry', 'last_name': 'Potter',
+             'wand_core': 'phoenix_feather'},
+            {'_id': 'c2', 'case_type': 'wizard', 'first_name': 'Tom', 'last_name': 'Riddle',
+             'wand_core': 'phoenix_feather'},
+            {'_id': 'c3', 'case_type': 'wizard', 'first_name': 'Ron', 'last_name': 'Weasley',
+             'wand_core': 'unicorn_hair'},
+        ]
+        self._assert_query_runs_correctly(
+            self.domain,
+            cases,
+            get_case_search_query(
+                self.domain,
+                ['wizard'],
+                {'_xpath_query': "OR(wand_core='phoenix_feather', last_name='Weasley')"},
+            ),
+            None,
+            ['c1', 'c2', 'c3']
+        )
+
+    def test_OR_one_positive_condition(self):
+        cases = [
+            {'_id': 'c1', 'case_type': 'wizard', 'first_name': 'Harry', 'last_name': 'Potter',
+             'wand_core': 'phoenix_feather'},
+            {'_id': 'c2', 'case_type': 'wizard', 'first_name': 'Tom', 'last_name': 'Riddle',
+             'wand_core': 'phoenix_feather'},
+            {'_id': 'c3', 'case_type': 'wizard', 'first_name': 'Ron', 'last_name': 'Weasley',
+             'wand_core': 'unicorn_hair'},
+        ]
+        self._assert_query_runs_correctly(
+            self.domain,
+            cases,
+            get_case_search_query(
+                self.domain,
+                ['wizard'],
+                {'_xpath_query': "OR(wand_core='dragon_heartstring', last_name='Weasley', '')"},
+            ),
+            None,
+            ['c3']
+        )
```

Comment:
Just adding my opinion, I'm also thinking no results with the same rationale that you said.

Reply:
Nope, that sounds great to me

Reply:
Just picking up on this thread - the JIRA ticket says "Exclude arguments that are just the empty string" which means `OR('', '') = OR()`. To me that suggests that if all conditions are empty the entire expression should be ignored which is equivalent to 'match all'.

I'm not sure exactly where this will be used by since it's an attempt to simplify long expressions by not having to determine which expressions to include, that the 'match all' case is the correct option.

Final thought - this decision should probably be confirmed with the delivery team.

Reply:
> `OR('', '') = OR()`
> Final thought - this decision should probably be confirmed with the delivery team.

I agree strongly on both points

> To me that suggests that if all conditions are empty the entire expression should be ignored which is equivalent to 'match all'.

I was initially uncertain, but I'm becoming more in favor of `match-none()` from a logical standpoint, as the arguments to the `OR` function are criteria for _inclusion_, so if none apply, then there's no rationale for including.  From a practical standpoint, however, I imagine desired behavior might vary depending on circumstances, and with `OR() = match-all()`, you could tweak the behavior to match none by adding `match-none()` to your args list (assuming we made that fn available), and it would only affect the base case:
```
// if OR() = match-all()
OR('', '')  -->  match-all()
OR(match-none(), '', '')  -->  match-none()
OR(match-none(), 'name="Ethan"', '')  ->  name="Ethan"

// if OR() = match-none()
OR('', '') --> match-none()
OR(match-all(), '', '')  -->  match-all()
OR(match-all(), 'name="Ethan"', '')  ->  match-all()  // this behavior is undesirable
```